### DISCUSSION TITLE
Error: Relay Protocol not supported: undefined in React Native app

### DISCUSSION THREAD

**Discussion by iGroza** [WalletConnect/walletconnect-monorepo#2302] [Developer Support]
## Issue Summary

When attempting to connect to the https://example.walletconnect.org demo using the Web3Wallet SDK package in my React Native application, I encounter the following error:

```
 {"context": "core"} {"context": "core/relayer"} {"context": "core/relayer/subscription"} [Error: Relay Protocol not supported: undefined]
```

## Environment

The following versions of the WalletConnect SDK are being used in my React Native app:

```
"@walletconnect/core": "2.7.2",
"@walletconnect/sign-client": "2.7.2",
"@walletconnect/utils": "2.7.2",
"@walletconnect/web3wallet": "1.7.0",
"@walletconnect/react-native-compat": "2.7.2"
```

My code of WalletConnect part in my React Native app is identical to the example in this repository: https://github.com/WalletConnect/web-examples/blob/main/wallets/react-web3wallet/src/utils/WalletConnectUtil.ts

To test my WalletConnect setup, I have used both https://example.walletconnect.org (v1) and https://react-app.walletconnect.com (v2).

## Expected behavior

I expect to be able to connect to the https://example.walletconnect.org demo without encountering any errors.

## Actual behavior

When attempting to connect to https://example.walletconnect.org, I encounter the error described above.

## Reproduction Steps

1. Open the [WalletConnectUtil.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-web3wallet/src/utils/WalletConnectUtil.ts) file from the WalletConnect web-examples repository
2. Copy the code and paste it into a React Native app that is using the  Web3Wallet SDK versions listed above
3. import `@walletconnect/react-native-compat` shims at main index.js file
4. Attempt to connect to the https://example.walletconnect.org demo using the app
5. Observe the error

## Additional Information

The repository mentioned above has a working web application at https://react-wallet.walletconnect.com/walletconnect, where both examples, https://example.walletconnect.org/ (v1) and https://react-app.walletconnect.com/ (v2), work without any errors. How can I achieve the same result in my React Native app?

**Comment by finessevanes** [comment]
Hey, thank you so much for all the information. https://example.walletconnect.org/ is actually a v1 dapp.

- **v1 URI**: wc:966a001a-d20f-41e8-a482-7fe6d548ee39@1?bridge=https%3A%2F%2F9.bridge.walletconnect.org&key=e5b72c727d1ecf62443d71982edf688b1aca5bc8be9857d26a264e07d2cc4be7
- **v2 URI:** wc:d86b1e810afa0b9aa7f516633f4c00c383f3839660ea5e3756ba22a61c4da3e3@2?relay-protocol=irn&symKey=a8d78dfec6482fe5d30d07d068d8ba11a5892661e1295797816d3b40526c5ba5

v2 will include the word `relay` and not `bridge`. Try testing your implementation with this test dapp: https://react-app.walletconnect.com/

From looking at your dependencies, you do not actually support v1.

**Reply by iPriyaranjan** [reply]
Hi @finessevanes, I want to integrate V1 and V2 together in my React Native app (similar to https://react-wallet.walletconnect.com/walletconnect) so that it works with both versions of DApps, can you suggest me How can I implement both the versions together in my React Native Wallet app. It would be nice if you share some example in react native.

**Reply by finessevanes** [reply]
We don't have examples of v1 and v2 running parallel for React Native. However, here are the [legacy docs](https://docs.walletconnect.com/1.0/quick-start/wallets/react-native) for v1 using React Native.

**Reply by crypblizz8** [reply]
@ignaciosantise can you assist here?

**Reply by ignaciosantise** [reply]
hey @iPriyaranjan 👋 as my teammate said, we don't have a wallet example with both versions. V1 will stop working in 3 weeks, and all DApps will have to migrate to V2 eventually, so i suggest you focus on make your wallet work well with V2.

You can check our v2 wallet examples here: https://github.com/WalletConnect/react-native-examples/tree/main/wallets 

and if you still want to implement v1, our [legacy docs](https://docs.walletconnect.com/1.0/quick-start/wallets/react-native) can help you with that
